src/components/org/apache/jmeter/extractor/json/jsonpath/JSONPostProcessor.java

```diff
@@ -0,0 +1,236 @@
+            jsonResponse = vars.get(getVariableName());
+            if(log.isDebugEnabled()) {
+                log.debug("JSON Extractor is using variable:"+getVariableName()+" which content is:"+jsonResponse);
+            }
+        } else {
+            SampleResult previousResult = context.getPreviousResult();
+            if (previousResult == null) {
+                return;
+            }
+            jsonResponse = previousResult.getResponseDataAsString();
+            if(log.isDebugEnabled()) {
+                log.debug("JSON Extractor processing result:"+jsonResponse);
+            }
+        }
+        String[] refNames = getRefNames().split(SEPARATOR);
+        String[] jsonPathExpressions = getJsonPathExpressions().split(SEPARATOR); 
+        String[] defaultValues = getDefaultValues().split(SEPARATOR);
+        int[] matchNumbers = getMatchNumbersAsInt(defaultValues.length);
+
+        //jsonResponse = jsonResponse.replaceAll("'", "\""); // $NON-NLS-1$  $NON-NLS-2$
+
+        if (refNames.length != jsonPathExpressions.length || 
+                refNames.length != defaultValues.length) {
+            log.error("Number of JSON Path variables must match number of default values and json-path expressions, check you use separator ';' if you have many values"); // $NON-NLS-1$
+            throw new IllegalArgumentException(JMeterUtils.getResString("jsonpp_error_number_arguments_mismatch_error")); // $NON-NLS-1$
+        } else {
+            for (int i = 0; i < jsonPathExpressions.length; i++) {
+                int matchNumber = matchNumbers[i];
+                String refNameFromRefNames = refNames[i].trim();
+                String jsonPathFromPathes = jsonPathExpressions[i].trim();
+                try {
+                    if(jsonResponse.isEmpty()) {
+                        vars.put(refNameFromRefNames, defaultValues[i]);
+                    } else {
+                        
+                        List<String> valuesExtracted = localMatcher.get().extractWithJsonPath(jsonResponse, jsonPathFromPathes);
+                        // if no values extracted, default value added
+                        if (valuesExtracted.isEmpty()) {
+                            vars.put(refNameFromRefNames, defaultValues[i]);
+                            vars.put(refNameFromRefNames+REF_MATCH_NR, "0");
+                            if(matchNumber<0 && getComputeConcatenation()) {
+                                log.debug("No value extracted, storing empty in :"+refNameFromRefNames + ALL_SUFFIX);
+                                vars.put(refNameFromRefNames + ALL_SUFFIX, "");
+                            }
+                        } else {
+                            // if more than one value extracted, suffix with
+                            // "_index"
+                            if (valuesExtracted.size() > 1) {
+                                if(matchNumber<0) {
+                                    int index = 0;
```

Comment:
We could start with `index = 1`, since we use it incremented only later (`index+1`)